### DISCUSSION TITLE
Let us pass in a express server directly!

### DISCUSSION THREAD

**Discussion by ManavGhaiCode** [socketio/socket.io#4135] [Ideas] [answered]
hello everyone, I am Manav Ghai a full stack dev. I was making a _discord clone_ for fun, and I used **socket.io** for live connection between two users on the platform. So, I use **_HTTP_ NodeJS** built in **library**, but I don't like the syntax of it over express. I know the way of using express with socket.io is -
`
const express = require('express');
const app = express();
const server = require('http').createServer(app);
const io = require('socket.io")(server);
`
...
`
server.listen(3000);
` 
This made me fell like braking my code, and it was making some errors. + My _"bace"_ boilerplate for a **_NodeJS_** with **_express_** is - 
`
const express = require('express');
const ejsMate = require('ejs-mate');
const path = require('path');

const app = express();

app.set('view engine', 'ejs');
app.set('views', path.join(__dirname, 'views'));

app.engine('ejs', ejsMate);

app.use(express.urlencoded({ extended: true }));
app.use(express.json());

app.get('/', (req, res) => {
    res.send('lol');
})

app.listen(3000, () => {
    console.log('serving on port 3000');
})
`

So, I think many other **devs** will be like this who don't like or know about **_HTTP_ NodeJS library**. + **express** syntax is just the best and we should keep it that way and mix it up with any other library's or replace it with any other  library's syntax. + I tried to find an other library and failed to find one just as good this one.

### So, think the following should happen -

1. **4.4.0alpha1 - _New version to change some thing like a emit should be a broadcast as the default_**.
2. **5.0.0rc1 - _New version as prototype for the 5.0.0 with the specified things above_**.
3. **5.0.0 - _New version the full version for the specified things above_**.

### Other thing which should happen -

1. **_A emit should be a broadcast as the default_**.
2. **_Less boilerplate code if possible_**.

**Comment by ManavGhaiCode** [comment]
Come on. Respond. Plzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz, this is the best idea you will get in your life man.

**Comment by darrachequesne** [comment, marked as answer]
Hi! You can totally reuse the httpServer that is returned by `app.listen()`:

```js
const express = require('express');
const ejsMate = require('ejs-mate');
const path = require('path');
const { Server } = require('socket.io');

const app = express();

app.set('view engine', 'ejs');
app.set('views', path.join(__dirname, 'views'));

app.engine('ejs', ejsMate);

app.use(express.urlencoded({ extended: true }));
app.use(express.json());

app.get('/', (req, res) => {
  res.send('lol');
})

const httpServer = app.listen(3000, () => {
  console.log('serving on port 3000');
})

const io = new Server(httpServer);

io.on('connection', () => {
  // ...
});
```

Regarding your other points:

> A emit should be a broadcast as the default.

If I understand correctly, you suggest that `socket.emit()` do a broadcast, is that right? Wouldn't that go against the `EventEmitter` abstraction between the server (`socket.emit(...)`) and the client (`socket.on(...)`)?

> Less boilerplate code if possible.

Sure, who wouldn't? :smile:  What do you suggest?

**Reply by ManavGhaiCode** [reply]
I think this fixes everything. thanks!
